setup.py

```diff
@@ -463,6 +463,9 @@ def setup_package():
     else:
         #from numpy.distutils.core import setup
         from setuptools import setup
+        # workaround for broken --no-build-isolation with newer setuptools,
+        # see gh-21288
+        metadata["packages"] = []
```

Comment:
Does this need a backport to 1.24.x?

Reply:
Sorry I missed this question. Yes, I think it's a good idea to backport this change. I'll open a PR now.

Reply:
(bit late, sorry about this. it can go into the next bug fix release)

Reply:
gh-22820